sources/optimizer/bcknd/device/hip/mma_kernel.h

```diff
@@ -36,69 +36,129 @@
 #define MMA_HIP_KERNEL_H
 
 template <typename T>
-__global__ void mma_Ljjxinv_kernel(T* __restrict__ Ljjxinv, 
+__global__ void delta_1dbeam_kernel(T* __restrict__ Delta,
+     const T L_total, const T Le, const int offset, const int n) {
+  int k = blockIdx.x * blockDim.x + threadIdx.x;
+  if (k >= n) return;
+
+  // Convert to 1-based indexing for the calculation
+  int idx = k + 1;
+
+  // Calculate first term: (L_total - Le*(offset+idx-1))^3
+  T x1 = L_total - Le * (T)(offset + idx - 1);
```

Comment:
If you need a cast then please write it explicitly
```suggestion
  T x1 = L_total - Le * static_cast<T>(offset + idx - 1);
```